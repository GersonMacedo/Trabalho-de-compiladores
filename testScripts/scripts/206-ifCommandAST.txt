! -s TREE
! should print correctly the AST with the if else commands
program p ;
    var b1 : boolean;
    var b2 : boolean;
    var b3 : boolean;
    var i1 : integer;
    var i2 : integer;
    var i3 : integer;
    var i4 : integer;
begin
    if b1 then
        i1 := 1
    ;

    if b2 then
        i2 := 3
    else
        i2 := 4
    ;

    if b1 then
        if b2 then
            i1 := 12
        else
            i2 := 5
    else
        if b3 then
            i3 := 23
        else
            i4 := 49
    ;
end .