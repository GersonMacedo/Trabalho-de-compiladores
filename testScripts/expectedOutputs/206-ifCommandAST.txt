
Performing up to syntactic analysis and printing the AST tree at testScripts/scripts/206-ifCommandAST.txt
No errors found

Programa 'p'
Declarações:
    'b1': BOOLEAN
    'b2': BOOLEAN
    'b3': BOOLEAN
    'i1': INTEGER
    'i2': INTEGER
    'i3': INTEGER
    'i4': INTEGER
Corpo:
    if
        'b1'
    then
        'i1' :=
            1
    if
        'b2'
    then
        'i2' :=
            3
    else
        'i2' :=
            4
    if
        'b1'
    then
        if
            'b2'
        then
            'i1' :=
                12
        else
            'i2' :=
                5
    else
        if
            'b3'
        then
            'i3' :=
                23
        else
            'i4' :=
                49
